oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LucenePropertyIndex.java

```diff
@@ -1237,13 +1238,18 @@ private static Query createQuery(String propertyName, PropertyRestriction pr,
                 } else if (pr.isNotNullRestriction()) {
                     // not null. For date lower bound of zero can be used
                     return NumericRangeQuery.newLongRange(propertyName, 0L, Long.MAX_VALUE, true, true);
+                } else if (pr.isNot && pr.not != null) {
+                    // -[property]=[value]
+                    BooleanQuery bool = new BooleanQuery();
+                    // This will exclude entries with [property]=[value]
+                    bool.add(NumericRangeQuery.newLongRange(propertyName, pr.not.getValue(LONG), pr.not.getValue(LONG), true, true), MUST_NOT);
```

Comment:
Except for the default case, the boolean queries are not returned. I think return statement is missing and the tests only test for strings, so these cases didn't get detected.

Reply:
good catch. Thanks. Fixed it now - https://github.com/apache/jackrabbit-oak/pull/465/commits/b45147efafc2fe77c2b576e1ebe1672426256838